### PR TITLE
Split ConfirmOrderJob to avoid blocking DJ worker

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

This unties these two email notifications so that they are picked up by a DJ worker independently. This should avoid the blocking the worker experiences (remember we still have a single one in all instances) when waiting between the two deliveries. See the flamegraph: 

Screenshot from 2021-05-27 13-01-53

https://app.datadoghq.com/apm/traces?end=1622015605459&paused=true&query=env%3Aproduction%20service%3Adelayed_job%20operation_name%3Adelayed_job%20resource_name%3AConfirmOrderJob%20%40duration%3A%3E%3D5s&start=1622009898303&streamTraces=true&trace=AQAAAXmngbg_woqc_QAAAABBWG1uZ2IwVkFBRHVDbWJkN25QTUVuY28&traceID=2916038355421570548&spanID=2005781139590273685.

Overall, both operations may take longer but other jobs can now be processed in between. Also, if any of the two fails, the other won't be affected.

#### What should we test?

There should be no change. Both emails should be received.

#### Release notes

Split ConfirmOrderJob and send order confirmation to shop and customer separately.
Changelog Category: Technical changes

### CODE REVIEW

**Review by andrewpbrett — APPROVED** [review]
Nice. Looks like a lazy class to me ;)

**Review by luisramos0 — APPROVED** [review]
Looks good but the build is broken.

File: app/models/spree/order.rb [hunk 1]
```diff
@@ -395,7 +395,8 @@ def finalize!
     def deliver_order_confirmation_email
       return if subscription.present?
 
-      ConfirmOrderJob.perform_later(id)
+      Spree::OrderMailer.confirm_email_for_customer(id).deliver_later
+      Spree::OrderMailer.confirm_email_for_shop(id).deliver_later
```
Comment (mkllnk) [inline app/models/spree/order.rb:399]: The specs say that we don't have `deliver_later`. Maybe we need to configure that first? We have just been using the Delayed::Job API until now.

File: spec/models/spree/order_spec.rb [hunk 2]
```diff
@@ -169,10 +169,16 @@
...
-        order.finalize!
-      end.to enqueue_job ConfirmOrderJob
+    it "sends confirmation emails to both the user and the shop owner" do
+      customer_confirm = double(:confirm_email_for_customer)
+      shop_confirm = double(:confirm_email_for_shop)
+
+      expect(Spree::OrderMailer).to receive(:confirm_email_for_customer).and_return(customer_confirm)
+      expect(Spree::OrderMailer).to receive(:confirm_email_for_shop).and_return(shop_confirm)
+      expect(customer_confirm).to receive(:deliver_later)
+      expect(shop_confirm).to receive(:deliver_later)
```
Comment (mkllnk) [inline spec/models/spree/order_spec.rb:179]: This test code is basically a duplication of the real code. You're just testing that you typed everything correctly but we are not testing that a job is added to the queue or that the job actually sends an email.

This spec passes even though our code fails. I think that we should test with real objects if possible.

**Review by sauloperez — COMMENTED** [review]
I did this in a bit of a rush and I was waiting to see the build's results. This is now ready. Feature specs cover the integration-level while the model specs cover the unit-level. Notice however, that trade-offs need to be made because two Rspec matches can't be combined chained together.

**Review by Matt-Yorkley — APPROVED** [review]
Yeaaahhhhh :sunglasses:

**Review by mkllnk — APPROVED** [review]

**Comment by Matt-Yorkley** [comment]
A quick rebase might be a good idea here...

**Comment by sauloperez** [comment]
done

**Comment by filipefurtad0** [comment]
Hey @sauloperez ,

I've staged and tested; both emails arrive, from:

- spree/order_mailer/confirm_email_for_shop.html.haml
- spree/order_mailer/_order_summary.html.haml

But I'm wondering if we need this PR now, since we've replaced DJ with sidekiq?

**Comment by mkllnk** [comment]
> But I'm wondering if we need this PR now, since we've replaced DJ with sidekiq?

This is still a good change.

* Less code.
* The failure of one email doesn't affect the other email.
* Emails can be sent in parallel.

**Comment by sauloperez** [comment]
:100: @mkllnk

**Pull request merged**